src/physics/nonlinear_solid.hpp

```diff
@@ -29,13 +29,31 @@ namespace serac {
  */
 class NonlinearSolid : public BasePhysics {
 public:
+  /**
+   * @brief A timestep method and config for the M solver
+   */
+  struct DynamicParameters {
+    TimestepMethod         timestepper;
+    LinearSolverParameters M_params;
+  };
+  /**
+   * @brief A configuration variant for the various solves
+   * Either quasistatic, or time-dependent with timestep and M params
+   */
+  struct NonlinearSolidParameters {
+    LinearSolverParameters           H_lin_params;
+    NonlinearSolverParameters        H_nonlin_params;
+    std::optional<DynamicParameters> dyn_params = std::nullopt;
+  };
+
```

Comment:
I would like clearer names here: `NonlinearSolid::DynamicParameters` sounds like it's related to the physics, not the method of solving an ODE. `NonlinearSolid::NonlinearSolidParameters` has a redundant prefix and also sounds like it should be contain the elasticity coefficients or something.

The might be combined by something like

```
class NonlinearSolid : public BasePhysics {
  struct SolverParameters {
    LinearSolverParameters                   linear;
    std::optional<NonlinearSolverParameters> nonlinear = std::nullopt;
    std::optional<TimestepMethod>            method = std::nullopt;
  };
  ...
};
```
or renamed to something like `{Dynamic,Quasistatic}-SolverParameters`

Reply:
I think this is a good idea - I've renamed to `<physics type>::SolverParameters` per your suggestion and renamed the extra dynamic solver parameters, when needed, to `<physics type>::DynamicSolverParameters`.  I think it's useful to keep the dynamic stuff together in an optional struct instead of optional fields to eliminate errors where one dynamic parameter is set but not the other.